lightning/src/ln/channelmanager.rs

```diff
@@ -1285,6 +1289,37 @@ impl<ChanSigner: ChannelKeys, M: Deref> ChannelManager<ChanSigner, M> where M::T
 		})
 	}
 
+	/// Generates a signed node_announcement from the given arguments and creates a
+	/// BroadcastNodeAnnouncement event.
+	///
+	/// RGB is a node "color" and alias a printable human-readable string to describe this node to
+	/// humans. They carry no in-protocol meaning.
+	///
+	/// addresses represent the set (possibly empty) of socket addresses on which this node accepts
```

Comment:
91f4e5d

Does it makes sense to pass a set containing both IPv*/onion addresses (maybe redundancy to get inbound connections if you don't want to rely only on port forwarding...) ? Not forbidding but maybe document somewhere the privacy implications.